src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+  // again
+  // Test 1: Middle of the capture window
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 2: Left of the capture window
+  x = 0;
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 3: Right of time graph
+  x = kTimeGraphSize[0];
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+}
+
+TEST_F(NavigationTestCaptureWindow, VerticalZoomWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+
+  float old_height = time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight();
+  MouseWheelMoved(x, y, 1, true);
+  UpdateChildrenPosAndSize();
+  EXPECT_GT(time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight(), old_height);
+
+  MouseWheelMoved(x, y, -1, true);
+  UpdateChildrenPosAndSize();
+  EXPECT_EQ(time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight(), old_height);
+}
+
+TEST_F(NavigationTestCaptureWindow, PanTimeWorksAsExpected) {
+  ExpectInitialState();
+
+  int x = 0;
+  int y = viewport_.GetScreenHeight() - 5;
+
+  // Pan time - need to zoom in a bit first, then pan slighty right and back again
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  KeyPressed('D', false, false, false);
+  UpdateChildrenPosAndSize();
+  EXPECT_GT(slider_->GetPosRatio(), 0.0);
+  EXPECT_GT(time_graph_->GetMinTimeUs(), 0.0);
+
+  KeyPressed('A', false, false, false);
+  UpdateChildrenPosAndSize();
+  EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+  EXPECT_EQ(time_graph_->GetMinTimeUs(), 0.0);
+}
+
+TEST_F(NavigationTestCaptureWindow, VerticalScrollWorksAsExpected) {
+  ExpectInitialState();
+
+  // Skipped for now until scrolling functionality is changed
```

Comment:
Nit: This should probably be a TODO and have a bug number. :innocent:

Reply:
I've actually removed this test now 😄 It should be part of the remaining work to change vertical scrolling.